### PR TITLE
Solve wrong enum name 

### PR DESCRIPTION
**Pull request by mcruzdev**
Fixes: https://github.com/quarkiverse/quarkus-openapi-generator/issues/495.

When an user configures an ENUM type with special characters like `/v1/users` the extension maps a enum with property `SLASH_V1/USERS`. This property is an invalid property and Java can not compile it.

Changes does:

- Add a new special characters into `QuarkusJavaClientCodegen` class.
- Override `toEnumVarName` and `initializeSpecialCharacterMapping` from  `AbstractJavaCodegen` class.
- Simplify `getSymbolName`that is responsible only for return "_symbol" symbol name when the symbol is on `specialCharReplacement` map.

The idea here was to "maintain" the same superclass behavior.

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-quarkus2` to backport the original PR to the `quarkus2` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java [hunk 1]
```diff
@@ -121,14 +121,50 @@ public void postProcess() {
...
+        if (" ".equals(enumValue)) {
+            return "SPACE";
+        }
+
+        if ("Integer".equals(datatype) || "Long".equals(datatype) ||
+                "Float".equals(datatype) || "Double".equals(datatype) || "BigDecimal".equals(datatype)) {
+            String varName = "NUMBER_" + enumValue;
+            varName = varName.replaceAll("-", "MINUS_");
+            varName = varName.replaceAll("\\+", "PLUS_");
+            varName = varName.replaceAll("\\.", "_DOT_");
```
Comment (hbelmiro) [inline deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:143]: We don't need to use regex here, right?

```suggestion
            varName = varName.replace("-", "MINUS_");
            varName = varName.replace("+", "PLUS_");
            varName = varName.replace(".", "_DOT_");
```

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java [hunk 2]
```diff
@@ -121,14 +121,50 @@ public void postProcess() {
...
+    @Override
+    protected String getSymbolName(String input) {
+        String symbolName = this.specialCharReplacements.get(input);
+        return symbolName != null ? symbolName.concat("_symbol") : null;
+    }
+
+    @Override
+    protected void initializeSpecialCharacterMapping() {
+        super.initializeSpecialCharacterMapping();
+        this.specialCharReplacements.put(" ", "Space");
```
Comment (fjtirado) [inline deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:168]: Do we need this one? 
I mean, I hardly see someone using just an empty space.

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java [hunk 3]
```diff
@@ -121,14 +121,50 @@ public void postProcess() {
     }
 
     @Override
-    protected String getSymbolName(String input) {
-        for (Entry<String, String> entry : specialCharReplacements.entrySet()) {
-            if (input.startsWith(entry.getKey())) {
-                return input.length() > entry.getKey().length()
-                        ? entry.getValue() + "_" + input.substring(entry.getKey().length())
-                        : entry.getValue() + "_symbol";
+    public String toEnumVarName(String value, String datatype) {
```
Comment (fjtirado) [inline deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:124]: Im wondering if we can call super and then add

        for (Map.Entry<String, String> entry : this.specialCharReplacements.entrySet()) {
            if (enumValue.contains(entry.getKey())) {
                enumValue = enumValue.replace(entry.getKey(), "_".concat(entry.getValue()).concat("_"));
            }
        }

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java [hunk 4]
```diff
@@ -121,14 +121,50 @@ public void postProcess() {
...
+            String varName = "NUMBER_" + enumValue;
+            varName = varName.replaceAll("-", "MINUS_");
+            varName = varName.replaceAll("\\+", "PLUS_");
+            varName = varName.replaceAll("\\.", "_DOT_");
+            return varName;
+        }
+
+        for (Map.Entry<String, String> entry : this.specialCharReplacements.entrySet()) {
+            if (enumValue.contains(entry.getKey())) {
+                enumValue = enumValue.replace(entry.getKey(), "_".concat(entry.getValue()).concat("_"));
```
Comment (fjtirado) [inline deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:149]: What happens if the special character is the last one?

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java [hunk 5]
```diff
@@ -121,14 +121,50 @@ public void postProcess() {
...
             }
         }
-        return null;
+
+        return enumValue
```
Comment (fjtirado) [inline deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:153]: This is an alternative implementation of super method section, which is the "effect" we are trying to achieve? 
```
// string
        String var = value.replaceAll("\\W+", "_").toUpperCase(Locale.ROOT);
        if (var.matches("\\d.*")) {
            return "_" + var;
        } else {
            return var;
        }
```

**Review by fjtirado — COMMENTED** [review]
Thanks a lot @mcruzdev 
I do not realize that any string starting with / will generated an invalid symbol (shame on me ;)
Overall looks good. Superb job. 
I added some comments which tries to minimize "copy paste" from parent class.

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java (hunk 4 above)
Comment (mcruzdev) [reply deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:149]: Hello, if the special character is the last one I replace with regex `_+` at final.

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java (hunk 2 above)
Comment (mcruzdev) [reply deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:168]: The user can create an enum as follow: `user name`, then I need to solve as `USER_SPACE_NAME`.

**Comment by mcruzdev** [comment]
> Thanks a lot @mcruzdev I do not realize that any string starting with / will generated an invalid symbol (shame on me ;) Overall looks good. Superb job. I added some comments which tries to minimize "copy paste" from parent class.

Hi @fjtirado TY for the review, I felt a pain when rewriting some parts of the super class method, I'm going to try to do this in a way where I reuse more of its code.

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java (hunk 1 above)
Comment (mcruzdev) [reply deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:143]: Hi @hbelmiro, TY!

I copied it from super, I will try to remove it in the next commits.

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java (hunk 2 above)
Comment (mcruzdev) [reply deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:168]: I put it to differentiate a " " (one space) from "  " (two spaces).

**Comment by mcruzdev** [comment]
Hi @fjtirado and @hbelmiro I am using the super method now, and to achieve this cases:

```java
"/status/addressStatus,String,SLASH_STATUS_SLASH_ADDRESSSTATUS",
  "$,String,DOLLAR_SYMBOL",
  "/users,String,SLASH_USERS",
  "'  ',String,EMPTY",
  "123456,String,_123456",
  "quarkus_resources,String,QUARKUS_RESOURCES",
  "123456,Integer,NUMBER_123456", // old behavior
  "123+123,Long,NUMBER_123PLUS_123" // old behavior
```

Was necessary to handle the result from super method. With this approach a enum `/users` (`USERS`) when mapped will be different than `users` (USERS). If you want to make more simple we can consider the result default from super class, what do you think?

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java [hunk 6]
```diff
@@ -121,14 +118,55 @@ public void postProcess() {
...
+
+        if (enumVarName.startsWith("NUMBER_")) {
+            return enumVarName;
+        }
+
+        Map<Integer, String> indexesOfSpecialChars = new TreeMap<>();
+        for (String key : this.specialCharReplacements.keySet()) {
+            // no consider underscore from super result
+            if (Objects.equals(key, "_"))
+                continue;
```
Comment (hbelmiro) [inline deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:141]:
```suggestion
            if (Objects.equals(key, "_")) {
                continue;
            }
```

File: deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java [hunk 7]
```diff
@@ -2,10 +2,7 @@
 
 import java.io.File;
 import java.net.URL;
-import java.util.List;
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.Optional;
+import java.util.*;
```
Comment (ricardozanini) [inline deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java:5]: Please don't use star imports. I think Quarkus has a code style config for every IDE, which we are using here.

**Review by ricardozanini — APPROVED** [review]

**Review by fjtirado — APPROVED** [review]

**Pull request merged**